### DISCUSSION TITLE
Is it possible to rename systematically output column from `agg`?

### DISCUSSION THREAD

**Discussion by yohplala** [vaexio/vaex#2204] [Q&A] [answered]
Hi,

[As per documentation](https://vaex.readthedocs.io/en/latest/tutorial.html#Group-by)
"when multiple aggregations are passed to a single column or expression, the output colums are appropriately named."
```python
df_pets.groupby(by='animal').agg({'age': 'mean',
                                  'cuteness': ['mean', 'std']})

#	animal	age	cuteness_mean	cuteness_std
0	dog	3.33333	9	0.816497
1	cat	5	5	0
2	guinea pig	2	6	2
```

Please, can we enforce this renaming systematically?
(even if there is a single aggregation function on a given column, that it systematically renamed into `colIn_aggFuncName`)

Or should I explicitly ask for renaming the output column, as described in the documentation as well.
"Also we can name the resulting aggregate columns as we wish."
```python
df_pets.groupby(by='animal',
                agg={'mean_age': vaex.agg.mean('age'),
                     'cuteness_unique_values': vaex.agg.nunique('cuteness'),
                     'cuteness_unique_min': vaex.agg.min('cuteness')})

#	animal	mean_age	cuteness_unique_values	cuteness_unique_min
0	dog	3.33333	3	8
1	cat	5	1	5
2	guinea pig	2	2	4
```

For this latter case, if I have to retain it, could we have a way, quite similar to pandas, to use function name (as string) instead of the function themselves?
In pandas, `agg` accepts this for instance `{col_out_name: (col_in_name, function_name)}`.

Thanks for your help.
Best regards,

**Comment by JovanVeljanoski** [comment, marked as answer]
I am quite happy with the way it is at the moment. With that said, I don't mind the proposal (the first code block). 
I would be weary of adding yet another API tho..  And in case of the pandas thing, that is a shortcut for `pd.NamedAgg` which i prefer to use in most serious cases.

I would live this one up to @maartenbreddels.

**Reply by yohplala** [reply]
Hi @JovanVeljanoski ,

> And in case of the pandas thing, that is a shortcut for pd.NamedAgg which i prefer to use in most serious cases.

Do you mean that vaex accepts pd.NamedAgg? Or that something similar exist un vaex?

**Reply by JovanVeljanoski** [reply]
>Do you mean that vaex accepts pd.NamedAgg? Or that something similar exist un vaex?

No, `pd.NamedAgg` is a pandas thing. No need for something like that to exist in vaex since we have the 2nd example you which is explicit and gives you all the options of the aggregations. It is more flexible then pandas, so I prefer it. It allows for selections, dropna, etc..
